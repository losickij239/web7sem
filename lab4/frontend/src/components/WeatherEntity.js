import React, {Component} from 'react';
import 'bootstrap/dist/css/bootstrap.css'

class WeatherEntity extends Component {

    render() {
        const {data} = this.props;

        return (
            <div className="alert alert-dark d-flex">
                <div>{data.type}</div>
                <div className="ml-auto">{data.value}</div>
            </div>
        )
    }

}

export default WeatherEntity