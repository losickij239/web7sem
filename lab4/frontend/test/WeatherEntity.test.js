import React from 'react';
import WeatherEntity from '../src/components/WeatherEntity';
import renderer from 'react-test-renderer';


const data = {
    type: 'Влажность',
    value: '123' + ' %'
};

it('отображается корректно', () => {
    const tree = renderer.create(
        <WeatherEntity data={data}/>
        ).toJSON();
    expect(tree).toMatchSnapshot();

});


