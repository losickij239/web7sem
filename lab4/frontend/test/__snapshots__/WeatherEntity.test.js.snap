// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`отображается корректно 1`] = `
<div
  className="alert alert-dark d-flex"
>
  <div>
    Влажность
  </div>
  <div
    className="ml-auto"
  >
    123 %
  </div>
</div>
`;
